aten/src/ATen/native/cpu/PowKernel.cpp

```diff
@@ -91,6 +91,62 @@ void pow_tensor_scalar_kernel(TensorIterator& iter, Scalar exp_scalar) {
         );
       }
     });
+  } else if (isComplexType(iter.dtype())) {
+    const auto exp = exp_scalar.to<std::complex<double>>();
```

Comment:
The copy paste here is irritating. What's the delta between tis and above?

Reply:
Yes, I know.
1. The most visible difference is 
``` C++
const auto exp = exp_scalar.to<double>();   //real
const auto exp = exp_scalar.to<std::complex<double>>(); // complex
```
Is exp cast to highest precision to maintain some numerical accuracy?

2. `std::pow()` was not very generous on various CI machines when base and exp were two different data types. I had to cast exp back to scalar_t when calling  the generic pow.
``` C++
[=](Vec base) -> Vec { return base.pow(exp);    //real
[=](Vec base) -> Vec { return base.pow(scalar_t(exp));  // complex
```

3. For complex numbers, the Vec256 optimizations of `sqrt()`, `rsqrt()` are tied to the performance of `std::log()`, not an AVX math function. I found a way to implement `log()` using AVX functions, but it was so complicated that it was slower than `std::log()`.  Two situations could play out in the future:
    -  There could be a need to add different accelerations to the pow kernel, because you pay a huge penalty for calling std::pow for complex numbers.
    -  Some features in AVX512 could dramatically speed up the complex implementation of `log()`, which would cause things to optimized differently. 

I didn't add any new accelleration cases to std::pow right now, but I could add them if you want. Eg. I can multiply base 10 times and it would still be faster than calling std::pow.

Reply:
> Is exp cast to highest precision to maintain some numerical accuracy?

In this case I think it is just convenience. Scalar stores the exponent internally as double and you have to extract it at that precision to make sure you don't lose any precision. But it's interesting that you need to downcast for the complex overloads.

Maybe we can add the comment here as a comment to the code in question.

Reply:
Yeah, I added the comment. Note the spec does say that it supports mixed complex precision.

```C++
template< class T, class U > 
complex</*Promoted*/> pow( const complex<T>& x, const complex<U>& y);
```

However when base is std::complex<float> and exp is std::complex<double> the code does not compile and it says there is no suitable overload of std::pow()???